app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt

```diff
@@ -86,10 +93,48 @@ class AdministratorControlsActivityPresenter @Inject constructor(
     getAdministratorControlsFragment()!!.setSelectedFragment(lastLoadedFragment)
     activity.supportFragmentManager.beginTransaction().add(
       R.id.administrator_controls_fragment_multipane_placeholder,
-      AppVersionFragment()
+      AppVersionFragment(),
+      ADMINISTRATOR_CONTROLS_BACKSTACK
     ).commitNow()
   }
 
+  fun handleloadProfileEdit(profileId: Int) {
+    if (!isMultipane) {
+      activity.startActivity(
+        ProfileEditActivity.createProfileEditActivity(
+          activity,
+          profileId,
+          isMultipane
+        )
+      )
+    } else {
+      lastLoadedFragment = PROFILE_EDIT_FRAGMENT
+      binding.administratorControlsMultipaneOptionsBackButton!!.visibility = View.VISIBLE
+      val fragment = ProfileEditFragment.newInstance(profileId, isMultipane)
+      activity.supportFragmentManager.beginTransaction().add(
+        R.id.administrator_controls_fragment_multipane_placeholder,
+        fragment,
+        ADMINISTRATOR_CONTROLS_BACKSTACK
+      ).commitNow()
+    }
+  }
+
+  val TAG = "tag"
+  fun handleOnBackPressed() {
+    var b = activity.supportFragmentManager.fragments.removeLast()
+    var c = activity.supportFragmentManager.fragments
+    Log.i(TAG, "handleOnBackPressed: $b\n$c")
+  }
+
+  fun handleOnResume() {
+    activity.supportFragmentManager.addOnBackStackChangedListener {
+      Log.i(
+        TAG,
+        "handleOnResume:\n **** back stack changed\n *******"
```

Comment:
logging if there is any change in the back stack in the log all the fragments displayed are added to the back stack properly but this output never comes very strange.